stock_archive_constraint/models/stock_location.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class StockLocation(models.Model):
+    _inherit = "stock.location"
+
+    @api.constrains("active")
+    def _check_active_stock_archive_constraint_stock_quant(self):
+        res = self.env['stock.quant'].search(
+            [
+                ('location_id', 'in', self.ids)
```

Comment:
This is not correct, as you are not going through children locations.

Reply:
Ok, changed now.

Reply:
As you discover the problems, it's good to add test use cases that covers these problems for avoiding regressions and have a good test suite.

Reply:
Added now more tests to check `child_of`